### PR TITLE
Remove kubectl's dependence on schema file in pkg/api/validation.

### PR DESCRIPTION
**Pull request by monopole**
**What this PR does / why we need it**:

Makes functions in validation/schema.go private to kubectl,
further isolating kubectl.  This move revealed a "hidden" dependence
(a dependence not expressed in a BUILD or make file) from a feature
level test in /hack/make-rules on a kubectl test data file.  So this
PR also adds some BUILD rules around the relevant hack targets, to make the
dependence official.  A later PR will move the kubectl aspect of this "hack"
test into a kubectl test directory.  Leaving it in place for now after establishing 
and "official" dependency, since moving the test beyond PR scope.  The
test also depends on a small sh file in //cluster, which makes no sense.

**Which issue this PR fixes**

Part of a series of PRs to address kubernetes/community#598

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by monopole** [comment]
/assign @lavalamp 

Need API owner's approval.

Please approve to unblock kubectl isolation work :)

**Comment by monopole** [comment]
looks like a test is annoyed by a moved data file.  will fix tomorrow, but meantime feel free to approve given that the merge won't happen till this is fixed.

**Comment by monopole** [comment]
@k8s-bot pull-kubernetes-federation-e2e-gce test this
@k8s-bot pull-kubernetes-unit test this

**Comment by pwittrock** [comment]
FYI, there is a test failure

**Comment by pwittrock** [comment]
@k8s-bot pull-kubernetes-node-e2e test this

**Comment by monopole** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by pwittrock** [comment]
@k8s-bot pull-kubernetes-node-e2e test this

**Comment by monopole** [comment]
/assign @mikedanese 
for cluster/BUILD

/assign @ixdy 
for build/BUILD

/assign @erictune 
for pkg/api/OWNERS

File: hack/make-rules/BUILD
Comment (ixdy) [inline hack/make-rules/BUILD]: test-cmd-util or test-cmd-lib might be a better name for this rule?

**Comment by mikedanese** [comment]
build and cluster changes:
/approve

File: hack/make-rules/BUILD
Comment (mikedanese) [inline hack/make-rules/BUILD]: Why did you make these sh_binary and not sh_test?

File: hack/make-rules/BUILD
Comment (monopole) [inline hack/make-rules/BUILD]: done

File: hack/make-rules/BUILD
Comment (monopole) [inline hack/make-rules/BUILD]: I'd like to, but it's too much work for this PR.
Modified the TODO to point to an issue against which the work can be done.

**Comment by mengqiy** [comment]
@k8s-bot pull-kubernetes-e2e-gce-etcd3 test this

**Comment by ixdy** [comment]
/approve
looks like this needs rebase, though.

**Comment by monopole** [comment]
Thanks, just going to wait till after the code freeze, and the massive types refactoring.

**Comment by pwittrock** [comment]
Hopefully @apelisse work will remove this dependency for us.

**Comment by monopole** [comment]
/assign @apelisse 

Antoine, PTAL, as this overlaps w/ your openapi work a bit.
all tests are passing

File: pkg/api/errors/BUILD
Comment (apelisse) [inline pkg/api/errors/BUILD]: Yeah, it's unfortunate you have to do that...

File: pkg/api/validation/validation_test.go
Comment (apelisse) [inline pkg/api/validation/validation_test.go]: OK I think that answers my question from this morning.

**Review by apelisse — APPROVED** [review]
All the lint-detected changes are super welcome.

Thanks for working on that, maybe I'd have liked a better separation of these multiples changes, but whatever

**Comment by apelisse** [comment]
Needs rebase :-/

**Comment by apelisse** [comment]
@monopole Feel free to self-lgtm after rebase.

**Comment by monopole** [comment]
@apelisse  self-lgtm? https://github.com/kubernetes/test-infra/blob/master/prow/plugins/lgtm/lgtm.go#L173

**Pull request merged**
